saleor/graphql/shop/mutations.py

```diff
@@ -288,3 +288,125 @@ def perform_mutation(cls, _root, info, key_type):
+        )
+
+    class Meta:
+        description = "Adds an staff notification."
+        permissions = ("site.manage_settings",)
+        error_type_class = ShopError
+        error_type_field = "shop_errors"
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        input_data = data["input"]
+
+        if "staff_email" not in input_data and "user" not in input_data:
+            raise ValidationError(
+                {
+                    "staff_notification": ValidationError(
+                        "User or staff email is required", code=ShopErrorCode.REQUIRED
+                    )
+                }
+            )
+        staff_notification_kwargs = {
+            "site_settings": info.context.site.settings,
+            "active": input_data.get("active", True),
+        }
+        if "user" in input_data:
+            user = cls.get_node_or_error(info, input_data["user"], only_type=User)
+            staff_notification_kwargs["user"] = user
+        elif "staff_email" in input_data:
+            staff_email = input_data["staff_email"]
+            staff_user = account_models.User.objects.filter(email=staff_email)
+            if staff_user.exists():
+                staff_notification_kwargs["user"] = staff_user.first()
+            else:
+                staff_notification_kwargs["staff_email"] = staff_email
+
+        instance = account_models.StaffNotification(**staff_notification_kwargs)
+        cls.clean_instance(instance)
+        instance.save()
+        return StaffNotificationAdd(staff_notification=instance)
+
+
+class StaffNotificationUpdate(BaseMutation):
+    staff_notification = graphene.Field(
+        StaffNotification, description="StaffNotification."
+    )
+
+    class Arguments:
+        id = graphene.ID(
+            required=True, description="ID of a StaffNotification to update."
+        )
+        input = StaffNotificationInput(
+            required=True, description="Fields required to update a product."
+        )
+
+    class Meta:
+        description = "Adds an staff notification."
+        permissions = ("site.manage_settings",)
+        error_type_class = ShopError
+        error_type_field = "shop_errors"
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        input_data = data["input"]
+        instance_id = data["id"]
+        instance = cls.get_node_or_error(info, instance_id, only_type=StaffNotification)
+
+        staff_notification_kwargs = {}
+        if "user" in input_data:
+            user = cls.get_node_or_error(info, input_data["user"], only_type=User)
+            instance.user = user
+            instance.staff_email = None
+            staff_notification_kwargs["user"] = user
+        elif "staff_email" in input_data:
+            staff_email = input_data["staff_email"]
+            staff_user = account_models.User.objects.filter(email=staff_email)
+            if staff_user.exists():
+                instance.user = staff_user.first()
```

Comment:
The same like above:
```suggestion
            staff_user = account_models.User.objects.filter(email=staff_email).first()
            if staff_user:
                instance.user = staff_user
```